integrations/user_test.go

```diff
@@ -18,3 +21,27 @@ func TestViewUser(t *testing.T) {
 	resp := MakeRequest(req)
 	assert.EqualValues(t, http.StatusOK, resp.HeaderCode)
 }
+
+func TestRenameInvalidUsername(t *testing.T) {
+	prepareTestEnv(t)
+
+	session := loginUser(t, "user2")
+	req := NewRequest(t, "GET", "/user/settings")
+	resp := session.MakeRequest(t, req)
+	assert.EqualValues(t, http.StatusOK, resp.HeaderCode)
+
+	htmlDoc := NewHTMLParser(t, resp.Body)
+	req = NewRequestWithValues(t, "POST", "/user/settings", map[string]string{
+		"_csrf": htmlDoc.GetCSRF(),
+		"name":  "%2f*", // not valid
```

Comment:
Added all invalid cases from #2066, and a reserved username test, and a valid username test.